lib/engine/action/reduce.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
```

Comment:
what is a reduce action

Reply:
i feel like you can just use the assign action for this

Reply:
It "reduces the cost of a private"

Reply:
yea, let's use that

Reply:
bca96f79017673bb5f853eb2899a76f486a7d7b8